runner.py

```diff
@@ -0,0 +1,163 @@
+import random
+
+class Game:
+    def __init__(self):
+        self.rooms = {
+            'coral reef': {
+                'description': 'You swim into a bright coral reef full of colorful fish.',
+                'exits': {'north': 'kelp forest', 'south': 'dark trench'},
+                'items': ['magical pearl'],
+                'characters': ['seahorse']
+            },
+            'dark trench': {
+                'description': 'You enter the dark trench. It\'s eerie and quiet.',
+                'exits': {'north': 'coral reef'},
+                'items': [],
+                'characters': ['shark']
+            },
+            'kelp forest': {
+                'description': 'You explore the kelp forest, swaying gently with the current.',
+                'exits': {'south': 'coral reef'},
+                'items': [],
+                'characters': []
+            }
+        }
+        self.current_room = 'coral reef'
+        self.inventory = []
+        self.characters = {
+            'seahorse': "The seahorse offers you a magical pearl that will help you on your journey.",
+            'shark': "The shark looks very menacing."
+        }
+        self.puzzle_solved = False
+
+    def print_intro(self):
+        print("Welcome to the Ocean!")
+        print("You are a wandering fish swimming in the vast ocean.")
+        print("Your journey will involve meeting interesting characters and making decisions that will shape your adventure.")
+        print("Let's dive in!")
+
+    def look(self):
+        room = self.rooms[self.current_room]
+        description = room['description']
+        items = ', '.join(room['items'])
+        characters = ', '.join(room['characters'])
+        print(f"{description}\nItems: {items}\nCharacters: {characters}")
+
+    def move(self, direction):
+        if direction in self.rooms[self.current_room]['exits']:
+            self.current_room = self.rooms[self.current_room]['exits'][direction]
+            self.look()
+        else:
+            print("You can't go that way.")
+
+    def examine(self, item):
+        if item in self.rooms[self.current_room]['items']:
+            print(f"You see a {item}.")
+        else:
+            print("There's no such item here.")
+
+    def take(self, item):
+        if item in self.rooms[self.current_room]['items']:
+            self.inventory.append(item)
+            self.rooms[self.current_room]['items'].remove(item)
+            print(f"You took the {item}.")
+        else:
+            print("There's no such item here.")
+
+    def use(self, item):
+        if item in self.inventory:
+            print(f"You used the {item}.")
+            # Implement item use logic if necessary
+        else:
+            print("You don't have that item.")
+
+    def talk_to(self, character):
```

Comment:
Character should also be it's own class! Similar to the comment I made about the rooms above.